tests/integration/test_export_inference.py

```diff
@@ -0,0 +1,255 @@
+def fxt_local_seed() -> int:
+    """The number of repetition for each test case.
+
+    The random seed will be set for [0, fxt_num_repeat - 1]. Default is one.
+    """
+    selected_seed = 7
+    msg = f"seed : {selected_seed}"
+    log.info(msg)
+    return selected_seed
+
+
+TASK_NAME_TO_MAIN_METRIC_NAME = {
+    "semantic_segmentation": "test/mIoU",
+    "multi_label_cls": "test/accuracy",
+    "multi_class_cls": "test/accuracy",
+}
+
+
+@pytest.mark.parametrize("recipe", RECIPE_LIST)
+def test_otx_export_infer(
+    recipe: str,
+    tmp_path: Path,
+    fxt_local_seed: int,
+    fxt_accelerator: str,
+    capfd: "pytest.CaptureFixture",
+) -> None:
+    """
+    Test OTX CLI e2e commands.
+
+    - 'otx train' with 2 epochs training
+    - 'otx test' with output checkpoint from 'otx train'
+    - 'otx export' with output checkpoint from 'otx train'
+    - 'otx test' with the exported model
+    - compare accuracy of the exported model vs the original accuracy
+
+    Args:
+        recipe (str): The recipe to use for training. (eg. 'classification/otx_mobilenet_v3_large.yaml')
+        tmp_path (Path): The temporary path for storing the training outputs.
+
+    Returns:
+        None
+    """
+    task = recipe.split("/")[-2]
+
+    if task not in TASK_NAME_TO_MAIN_METRIC_NAME or "dino_v2" in recipe:
+        pytest.skip(f"Inference pipeline for {recipe} is not implemented")
+
+    # litehrnet_* models don't support deterministic mode
+    model_name = recipe.split("/")[-1].split(".")[0]
+    deterministic_flag = "False" if "litehrnet" in recipe else "True"
+
+    # 1) otx train
+    tmp_path_train = tmp_path / f"otx_train_{model_name}"
+    command_cfg = [
+        "otx",
+        "train",
+        "--config",
+        recipe,
+        "--data_root",
+        DATASET[task]["data_root"],
+        "--engine.work_dir",
+        str(tmp_path_train / "outputs"),
+        "--engine.device",
+        fxt_accelerator,
+        "--max_epochs",
+        "2",
+        "--seed",
+        f"{fxt_local_seed}",
+        "--deterministic",
+        deterministic_flag,
+        *DATASET[task]["overrides"],
+    ]
+
+    with patch("sys.argv", command_cfg):
+        main()
+
+    ckpt_files = list((tmp_path_train / "outputs" / "checkpoints").glob(pattern="epoch_*.ckpt"))
+    assert len(ckpt_files) > 0
+
+    # 2) otx test
+    tmp_path_test = tmp_path / f"otx_test_{model_name}"
+    command_cfg = [
+        "otx",
+        "test",
+        "--config",
+        recipe,
+        "--data_root",
+        DATASET[task]["data_root"],
+        "--engine.work_dir",
+        str(tmp_path_test / "outputs"),
+        "--engine.device",
+        fxt_accelerator,
+        *DATASET[task]["overrides"],
+        "--checkpoint",
+        str(ckpt_files[-1]),
+    ]
+
+    with patch("sys.argv", command_cfg):
+        main()
+
+    # 3) otx export
+    format_to_ext = {"OPENVINO": "xml"}  # [TODO](@Vlad): extend to "ONNX": "onnx"
```

Comment:
It's just a question. Are there any reasons not to test ONNX now?

Reply:
Because onnx inference is not fully supported yet. Ability to infer to an onnx file is checked in cli e2e tests

Reply:
yes but don't we need to check whether model can be exported to ONNX?

Reply:
It's done in `tests/integration/cli/test_cli.py`